packages/google_sign_in/google_sign_in_web/lib/src/utils.dart

```diff
@@ -2,59 +2,89 @@
 
-import 'dart:async';
-import 'dart:html' as html;
+import 'dart:convert';
 
+import 'package:google_identity_services_web/id.dart';
+import 'package:google_identity_services_web/oauth2.dart';
 import 'package:google_sign_in_platform_interface/google_sign_in_platform_interface.dart';
 
-import 'js_interop/gapiauth2.dart' as auth2;
-
-/// Injects a list of JS [libraries] as `script` tags into a [target] [html.HtmlElement].
-///
-/// If [target] is not provided, it defaults to the web app's `head` tag (see `web/index.html`).
-/// [libraries] is a list of URLs that are used as the `src` attribute of `script` tags
-/// to which an `onLoad` listener is attached (one per URL).
-///
-/// Returns a [Future] that resolves when all of the `script` tags `onLoad` events trigger.
-Future<void> injectJSLibraries(
-  List<String> libraries, {
-  html.HtmlElement? target,
-}) {
-  final List<Future<void>> loading = <Future<void>>[];
-  final List<html.HtmlElement> tags = <html.HtmlElement>[];
-
-  final html.Element targetElement = target ?? html.querySelector('head')!;
-
-  for (final String library in libraries) {
-    final html.ScriptElement script = html.ScriptElement()
-      ..async = true
-      ..defer = true
-      // ignore: unsafe_html
-      ..src = library;
-    // TODO(ditman): add a timeout race to fail this future
-    loading.add(script.onLoad.first);
-    tags.add(script);
+/// A codec that can encode/decode JWT payloads.
+///
+/// See https://www.rfc-editor.org/rfc/rfc7519#section-3
+final Codec<Object?, String> jwtCodec = json.fuse(utf8).fuse(base64);
+
+/// A RegExp that can match, and extract parts from a JWT Token.
+///
+/// A JWT token consists of 3 base-64 encoded parts of data separated by periods:
+///
+///   header.payload.signature
+///
+/// More info: https://regexr.com/789qc
+final RegExp jwtTokenRegexp = RegExp(
+    r'^(?<header>[^\.\s]+)\.(?<payload>[^\.\s]+)\.(?<signature>[^\.\s]+)$');
+
+/// Decodes the `claims` of a JWT token and returns them as a Map.
+///
+/// JWT `claims` are stored as a JSON object in the `payload` part of the token.
+///
+/// (This method does not validate the signature of the token.)
+///
+/// See https://www.rfc-editor.org/rfc/rfc7519#section-3
+Map<String, Object?>? getJwtTokenPayload(String? token) {
+  if (token != null) {
+    final RegExpMatch? match = jwtTokenRegexp.firstMatch(token);
+    if (match != null) {
+      return decodeJwtPayload(match.namedGroup('payload'));
+    }
   }
 
-  targetElement.children.addAll(tags);
-  return Future.wait(loading);
+  return null;
 }
 
-/// Utility method that converts `currentUser` to the equivalent [GoogleSignInUserData].
+/// Decodes a JWT payload using the [jwtCodec].
+Map<String, Object?>? decodeJwtPayload(String? payload) {
+  try {
+    // Payload must be normalized before passing it to the codec
+    return (jwtCodec.decode(base64.normalize(payload!))
+            as Map<String, Object?>?)
+        ?.cast<String, Object?>();
```

Comment:
It looks like this cast is the same as the `as` cast just above, which should mean that it's not needed (unless I'm mistaken about the strictness of `as`).

Reply:
Ah yes, I totally cargo-culted this (wrong) from the google maps' convert.dart file.